lightning/src/util/enforcing_trait_impls.rs

```diff
@@ -71,6 +96,19 @@ impl ChannelKeys for EnforcingChannelKeys {
 	}
 
 	fn sign_local_commitment<T: secp256k1::Signing + secp256k1::Verification>(&self, local_commitment_tx: &LocalCommitmentTransaction, secp_ctx: &Secp256k1<T>) -> Result<Signature, ()> {
+		let revoked = self.revoked_commitment.lock().unwrap();
+		let keys = &local_commitment_tx.local_keys;
+		if keys.per_commitment_point != self.inner.get_per_commitment_point(*revoked - 1, secp_ctx) {
+			if keys.per_commitment_point != self.inner.get_per_commitment_point(*revoked - 2, secp_ctx) {
+				if keys.per_commitment_point == self.inner.get_per_commitment_point(*revoked, secp_ctx) {
+					panic!("attempted to sign the latest revoked local commitment {}", self.inner.commitment_seed[0]);
+				} else {
+					panic!("can only sign the next two unrevoked commitment numbers, {} revoked={} point={}",
```

Comment:
All the tests pass without this panic, and, indeed, I think you are correct in that it is just different objects getting serialized/read breaking things in the tests. It would be nice to enforce the other new assertions upstream even if we can't hit this one, but I think the solution is going to have to be some global static HashMap tracking revocation for a given commitment seed.